usaspending_api/common/helpers/fiscal_year_helpers.py

```diff
@@ -185,38 +184,17 @@ def bolster_missing_time_periods(filter_time_periods, queryset, date_range_type,
     return results
 
 
-def calculate_last_completed_fiscal_quarter(fiscal_year, as_of_date=current_fiscal_date()):
+def calculate_last_completed_fiscal_quarter(fiscal_year):
     """
-    ENABLE_CARES_ACT_FEATURES TECH DEBT:  Make this work with new standardized, yet-to-be-named
-    function.  There are currently several flavors flying around.  Waiting for one to land rather
-    than creating another.
-
     Returns either the most recently completed fiscal quarter or None if it's too early in the
     fiscal year for the first quarter to be considered "completed".  Should always return None for
-    future fiscal years.  as_of_date was intended for unit testing purposes, but who knows, maybe
-    you'll find another use for it being the unquestionable genius that you are.
+    future fiscal years.
     """
 
-    # Get the current fiscal year so that it can be compared against the FY in the request.
-    day_difference = as_of_date - timedelta(days=SUBMISSION_WINDOW_DAYS)
-    current_fiscal_date_adjusted = FiscalDateTime(day_difference.year, day_difference.month, day_difference.day)
-
-    if fiscal_year == current_fiscal_date_adjusted.fiscal_year:
-        current_fiscal_quarter = current_fiscal_date_adjusted.quarter
-        # If it's currently the first quarter (or within SUBMISSION_WINDOW_DAYS days of the
-        # first quarter), return None because it's too soon for there to be a completed quarter
-        # for fiscal_year.
-        if current_fiscal_quarter == 1:
-            fiscal_quarter = None
-        else:
-            fiscal_quarter = current_fiscal_quarter - 1
-    elif fiscal_year < current_fiscal_date_adjusted.fiscal_year:
-        fiscal_quarter = 4
-    else:
-        # The future cannot have completed quarters unless you're into shady accounting.
-        fiscal_quarter = None
-
-    return fiscal_quarter
+    row = DABSSubmissionWindowSchedule.objects.filter(
```

Comment:
For previous years it works as expected by providing `fiscal_quarter = 4`. However, if the `fiscal_year = <CURRENT FISCAL YEAR>` then it is returning the last `fiscal_quarter` even if it is not completed. I think there is a bug in the current code that is also not handling the case of returning the last completed quarter. Stepping through the old implementation the `as_of_date` is the `current_fiscal_date` (or today + 3 months). Then the FiscalQuarter inside of FiscalDateTime is also adjusting for 3 months.

This comment is what I am referring to by returning the last completed quarter:
```
Returns either the most recently completed fiscal quarter or None if it's too early in the
fiscal year for the first quarter to be considered "completed".
```

SQL for current implementation: 
```
SELECT 
	MAX("dabs_submission_window_schedule"."submission_fiscal_quarter") AS "submission_fiscal_quarter__max"
FROM 
	"dabs_submission_window_schedule"
WHERE 
	"dabs_submission_window_schedule"."submission_reveal_date" <= '2021-03-19T13:57:27.109797+00:00'::timestamptz 
	AND "dabs_submission_window_schedule"."submission_fiscal_year" = 2021;
```

Reply:
@sethstoudenmier So if nothing matches the two filters below, it returns None as expected (confirmed in tests). Maybe we should talk? I'm not understanding the problem you're pointing out.

Reply:
Followed up in Slack pertaining to this. Outcome is to look at using the is_quarter column to avoid returning a quarter that has only a single period revealed.